### ISSUE TITLE
fastlane match hangs forever when TERM is 'dumb'

### ISSUE THREAD

**Issue by ojechev-broadsoft** [fastlane/fastlane#11732]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [ ] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Hi Team!

After I've upgraded build nodes for some reason 'fastlane match' started to fail - hangs forever.
Build nodes are running OSX Sierra (10.12.6) and are connected to Jenkins master using swarm plugin.

The problem occurs only when the build job is started from Jenkins. I had no problem when executing commands over SSH session or directly on Terminal using X session.

Finally I was able to find the difference. Verbose output shew the message:

```
INFO [2018-01-29 19:16:03.81]: $ git branch --list origin/branch --no-color -r
INFO [2018-01-29 19:16:03.81]: ▸ WARNING: terminal is not fully functional
```
'ps -ax' shows that this process is active. When I kill it the job execution  is unblocked and few more lines are added to the output that could be useful for you:

```
00:01:42.272 INFO [2018-01-29 19:17:20.02]: ▸ -  (press RETURN)
00:01:42.273 WARNING: terminal is not fully functional
00:01:42.273 -  (press RETURN)
```
The rest of the output is not related to the issue and just shows that the job execution was interrupted.

I added line to print the TERM and it printed **dumb**. 
I checked the value on my SSH session and on my X session and it was **xterm-256color**.

### The workaround
So I've added line in my build script to set the TERM to 'xterm-256color' and then 'fastlane match' was able to complete successfully!

Not sure what exactly cause the issue to appear as I've updated several components: git, ruby, fastlane, gem.

Is this is an expected behavior or not?
I spent few hours chasing the issue. I hope that by posting this issue will help somebody to fix his CI faster.

Let me know if you need more information and I will provide it.

##### Complete output when running fastlane, including the stack trace and command used

00:00:20.120 INFO [2018-01-29 19:15:57.87]: Cloning remote git repo...
00:00:20.120 INFO [2018-01-29 19:15:57.87]: $ GIT_TERMINAL_PROMPT=0 git clone 'ssh://git@<repo>' '/var/folders/hr/y1nk4zfn7yvcn9jwjvmfqmc80000gn/T/d20180129-14734-1xo5pra' -b <branch> --single-branch
00:00:20.125 INFO [2018-01-29 19:15:57.87]: ▸ Cloning into '/var/folders/hr/y1nk4zfn7yvcn9jwjvmfqmc80000gn/T/d20180129-14734-1xo5pra'...
00:00:21.598 INFO [2018-01-29 19:15:59.35]: ▸ remote: Counting objects: 799, done.
00:00:21.599 INFO [2018-01-29 19:15:59.35]: ▸ remote: Compressing objects: 100% (4/4), done.
00:00:25.904 INFO [2018-01-29 19:16:03.66]: ▸ remote: Total 799 (delta 9), reused 8 (delta 8), pack-reused 787
00:00:25.905 INFO [2018-01-29 19:16:03.66]: ▸ Receiving objects: 100% (799/799), 4.27 MiB | 1.02 MiB/s, done.
00:00:25.908 INFO [2018-01-29 19:16:03.66]: ▸ Resolving deltas: 100% (193/193), done.
00:00:26.053 INFO [2018-01-29 19:16:03.81]: $ git branch --list origin/<branch> --no-color -r
00:00:26.061 INFO [2018-01-29 19:16:03.81]: ▸ WARNING: terminal is not fully functional

### Environment

<details><summary>✅ fastlane environment ✅</summary>
00:00:21.169 
00:00:21.169 ### Stack
00:00:21.169 
00:00:21.169 | Key                         | Value                                           |
00:00:21.170 | --------------------------- | ----------------------------------------------- |
00:00:21.170 | OS                          | 10.12.6                                         |
00:00:21.170 | Ruby                        | 2.5.0                                           |
00:00:21.170 | Bundler?                    | false                                           |
00:00:21.170 | Git                         | git version 2.16.1                              |
00:00:21.170 | Installation Source         | /usr/local/bin/fastlane                         |
00:00:21.170 | Host                        | Mac OS X 10.12.6 (16G1212)                      |
00:00:21.170 | Ruby Lib Dir                | /usr/local/Cellar/ruby/2.5.0/lib                |
00:00:21.170 | OpenSSL Version             | OpenSSL 1.0.2n  7 Dec 2017                      |
00:00:21.171 | Is contained                | false                                           |
00:00:21.171 | Is homebrew                 | false                                           |
00:00:21.171 | Is installed via Fabric.app | false                                           |
00:00:21.171 | Xcode Path                  | /Applications/XCode-9.2.app/Contents/Developer/ |
00:00:21.171 | Xcode Version               | 9.2                                             |
00:00:21.171 
00:00:21.171 
00:00:21.171 ### System Locale
00:00:21.171 
00:00:21.171 | Variable | Value       |   |
00:00:21.171 | -------- | ----------- | - |
00:00:21.171 | LANG     | en_US.UTF-8 | ✅ |
00:00:21.171 | LC_ALL   | en_US.UTF-8 | ✅ |
00:00:21.171 | LANGUAGE |             |   |
00:00:21.175 ### fastlane gems
00:00:21.175 
00:00:21.175 | Gem      | Version | Update-Status |
00:00:21.175 | -------- | ------- | ------------- |
00:00:21.175 | fastlane | 2.78.0  | ✅ Up-To-Date  |
<details><summary><b>Loaded gems</b></summary>
00:00:21.176 
00:00:21.176 | Gem                       | Version      |
00:00:21.176 | ------------------------- | ------------ |
00:00:21.176 | did_you_mean              | 1.2.0        |
00:00:21.176 | slack-notifier            | 2.3.2        |
00:00:21.176 | CFPropertyList            | 2.3.6        |
00:00:21.176 | claide                    | 1.0.2        |
00:00:21.176 | colored2                  | 3.1.2        |
00:00:21.176 | nanaimo                   | 0.2.3        |
00:00:21.176 | xcodeproj                 | 1.5.4        |
00:00:21.176 | rouge                     | 2.0.7        |
00:00:21.176 | xcpretty                  | 0.2.8        |
00:00:21.176 | terminal-notifier         | 1.8.0        |
00:00:21.176 | unicode-display_width     | 1.3.0        |
00:00:21.176 | terminal-table            | 1.8.0        |
00:00:21.177 | plist                     | 3.4.0        |
00:00:21.177 | addressable               | 2.5.2        |
00:00:21.177 | multipart-post            | 2.0.0        |
00:00:21.177 | word_wrap                 | 1.0.0        |
00:00:21.177 | public_suffix             | 2.0.5        |
00:00:21.177 | tty-screen                | 0.6.4        |
00:00:21.177 | tty-cursor                | 0.5.0        |
00:00:21.177 | tty-spinner               | 0.8.0        |
00:00:21.177 | babosa                    | 1.0.2        |
00:00:21.177 | colored                   | 1.2          |
00:00:21.177 | highline                  | 1.7.10       |
00:00:21.177 | commander-fastlane        | 4.4.5        |
00:00:21.177 | excon                     | 0.60.0       |
00:00:21.177 | faraday                   | 0.14.0       |
00:00:21.177 | unf_ext                   | 0.0.7.4      |
00:00:21.177 | unf                       | 0.1.4        |
00:00:21.178 | domain_name               | 0.5.20170404 |
00:00:21.178 | http-cookie               | 1.0.3        |
00:00:21.178 | faraday-cookie_jar        | 0.0.6        |
00:00:21.178 | fastimage                 | 2.1.1        |
00:00:21.178 | gh_inspector              | 1.0.3        |
00:00:21.178 | json                      | 2.1.0        |
00:00:21.178 | mini_magick               | 4.5.1        |
00:00:21.178 | multi_json                | 1.13.1       |
00:00:21.178 | multi_xml                 | 0.6.0        |
00:00:21.178 | rubyzip                   | 1.2.1        |
00:00:21.178 | security                  | 0.1.3        |
00:00:21.178 | xcpretty-travis-formatter | 1.0.0        |
00:00:21.178 | dotenv                    | 2.2.1        |
00:00:21.178 | faraday_middleware        | 0.12.2       |
00:00:21.178 | uber                      | 0.1.0        |
00:00:21.178 | declarative               | 0.0.10       |
00:00:21.178 | declarative-option        | 0.1.0        |
00:00:21.179 | representable             | 3.0.4        |
00:00:21.179 | retriable                 | 3.1.1        |
00:00:21.179 | mime-types-data           | 3.2016.0521  |
00:00:21.179 | mime-types                | 3.1          |
00:00:21.179 | little-plugger            | 1.1.4        |
00:00:21.179 | logging                   | 2.2.2        |
00:00:21.179 | jwt                       | 2.1.0        |
00:00:21.179 | memoist                   | 0.16.0       |
00:00:21.179 | os                        | 0.9.6        |
00:00:21.179 | signet                    | 0.8.1        |
00:00:21.179 | googleauth                | 0.6.2        |
00:00:21.179 | httpclient                | 2.8.3        |
00:00:21.179 | google-api-client         | 0.13.6       |
00:00:21.179 | bundler                   | 1.16.1       |
00:00:21.179 | openssl                   | 2.1.0        |
00:00:21.179 | date                      | 1.0.0        |
00:00:21.180 | fileutils                 | 1.0.2        |
00:00:21.180 | ipaddr                    | 1.2.0        |
00:00:21.180 </details>
00:00:21.180 
00:00:21.180 
00:00:21.180 *generated on:* **2018-01-29**

**Comment by taquitos** [comment]
This is great information to have, thank you for documenting it!

We do some output redirection, so I'm not surprised if the terminal isn't set to something we consider "sane", things might break.

**Comment by atreat** [comment]
I'm running into this on CircleCi 2.0 as well

**Comment by joshdholtz** [comment]
@atreat So did this end up solving it completely for you?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by atreat** [comment]
@joshdholtz yup looks like it did.
